src/go/rpk/pkg/kafka/client_franz.go

```diff
@@ -93,17 +95,27 @@ func NewFranzClient(fs afero.Fs, p *config.RpkProfile, extraOpts ...kgo.Opt) (*k
 	}
 
 	if k.SASL != nil {
-		mech := scram.Auth{
-			User: k.SASL.User,
-			Pass: k.SASL.Password,
-		}
-		switch name := strings.ToUpper(k.SASL.Mechanism); name {
-		case "SCRAM-SHA-256", "": // we default to SCRAM-SHA-256 -- people commonly specify user & pass without --sasl-mechanism
-			opts = append(opts, kgo.SASL(mech.AsSha256Mechanism()))
-		case "SCRAM-SHA-512":
-			opts = append(opts, kgo.SASL(mech.AsSha512Mechanism()))
-		default:
-			return nil, fmt.Errorf("unknown SASL mechanism %q, supported: [SCRAM-SHA-256, SCRAM-SHA-512]", name)
+		if k.SASL.Mechanism == admin.CloudOIDC {
+			a := p.CurrentAuth()
+			if a == nil || a.AuthToken == "" {
+				return nil, fmt.Errorf("please login to our cloud using 'rpk cloud login'") // TODO
+			}
+			opts = append(opts, kgo.SASL((oauth.Auth{
+				Token: a.AuthToken,
+			}).AsMechanism()))
+		} else {
+			mech := scram.Auth{
```

Comment:
... while this is a word when both live briefly in scope?

Reply:
Changing to `a`